src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,12 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null || user.getLogin() == null || user.getId() == null
+                || user.getAge() == null || user.getPassword() == null
+                || user.equals(storageDao.get(user.getLogin()))
+                || user.getAge() < 18 || user.getPassword().length() < 6) {
+            throw new RuntimeException("Cannot register user");
+        }
+        return storageDao.add(user);
```

Comment:
It's easier to understand your code, if each check will be in  separate methods or steps